src/shred_network/shred_receiver.zig

```diff
@@ -41,68 +42,102 @@ pub const ShredReceiver = struct {
 
     const Self = @This();
 
+    /// A shared instance of an event to support waiting on multiple Channels.
+    const ReceiverSignal = struct {
```

Comment:
I can see now why you envisioned all signals being hooks.

What do you think about having the channels share the same ResetEvent? For example you could change this Channel field to a pointer:
```
event: *std.Thread.ResetEvent = .{},
```

Reply:
Since it's only this case, decided to make the `turbine_socket` & `repair_socket` run their own `runPacketHandler` threads for now (instead of having them share the same thread + ReceiveEvent hook).

https://github.com/Syndica/sig/pull/486/commits/84836dda2f90eac80aa474ae4ded2c704ef08f2b